ng2-material/components/paging/paging.ts

```diff
@@ -0,0 +1,286 @@
+
+      let rest = this.model.totalItems - _rangeStart,
+        _rangeStop = rest < this.model.itemsPerPage ? this.model.totalItems : _rangeStart + this.model.itemsPerPage - 1;
+
+      return this._getFormattedValue(_rangeStart, _rangeStop, this.model.totalItems);
+    }
+
+    return;
+  }
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.paging;
+      });
+  }
+}
+
+@Component({
+  selector: 'md-paging-controls',
+  template: `
+      <span [class.md-paging-control-active]="!isFirstPage()" class="md-paging-control md-paging-control-previous">
+        <button (click)="previousPage()" class="material-icons">keyboard_arrow_left</button>
+      </span>
+      <span [class.md-paging-control-active]="!isLastPage()" class="md-paging-control md-paging-control-next">
+        <button (click)="nextPage()" class="material-icons">keyboard_arrow_right</button>
+      </span>
+    `,
+  host: {
+    '[class.md-paging-controls]': 'true'
+  }
+})
+export class MdPagingControls {
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPagingModel;
+
+  isFirstPage() {
+    return isPresent(this.model) && this.model.currentPage == 1;
+  }
+
+  isLastPage() {
+    return isPresent(this.model) && this.model.totalItems <= this.model.currentPage * this.model.itemsPerPage;
+  }
+
+  previousPage() {
+    this._changePage(this.model.currentPage - 1);
+  }
+
+  nextPage() {
+    this._changePage(this.model.currentPage + 1);
+  }
+
+  _changePage(newPage: number) {
+    let _model = JSON.parse(JSON.stringify(this.model));
+    _model.currentPage = newPage;
+    this._service.change(this.name, _model);
+  }
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.paging;
+      });
+  }
+}
+
+@Component({
+  selector: 'md-paging-length-selector',
+  template: `
+   <span *ngIf="selectorLengthBefore" class="md-paging-length-selector-label md-paging-length-selector-before">{{selectorLengthBefore}}</span>
+   <select [(ngModel)]="model.itemsPerPage" (ngModelChange)="changePagingLength($event)" class="md-paging-length-select">
+      <option *ngFor="#length of selectorLengthAvailable" [value]="length">
+        {{length}}
+      </option>
+    </select>
+    <span *ngIf="selectorLengthAfter" class="md-paging-length-selector-label md-paging-length-selector-after">{{selectorLengthAfter}}</span>
+  `,
+  host: {
+    '[class.md-paging-length-selector]': 'true',
+    '[hidden]': '!selectorLengthAvailable.length'
+  }
+})
+export class MdPagingLengthSelector {
+  @Input()
+  selectorLengthBefore: string = 'Rows per page:';
```

Comment:
I don't think the attributes/properties on a component need to contain part of the component name to be clear, and it tends to confuse the attribute meaning, imo. For example the `MdPageLengthSelector` can have `lengthOptions`, `lengthAvailable`, `lenghtOptions`, etc. Since they become attributes on the `md-paging-length-selector` tag, I don't think they need to be prefixed - though I guess they should be if it were an attribute directive instead.